lib/Data.php

```diff
@@ -347,11 +350,26 @@ public function validateFilter($filterValue) {
 	 */
 	public function expire($expireDays = 365) {
 		$ttl = (60 * 60 * 24 * max(1, $expireDays));
-
 		$timelimit = time() - $ttl;
-		$this->deleteActivities([
+		$conditions = [
 			'timestamp' => [$timelimit, '<'],
-		]);
+		];
+
+		$excludedUsers = $this->config->getSystemValue('activity_expire_exclude_users', []);
+		if (!empty($excludedUsers)) {
+			foreach ($excludedUsers as $user) {
+				$condition1 = [
+					'user' => [$user, '!=']
```

Comment:
should be only the `affecteduser` below.
This here would mean that actions of the given user would never expire.

Also this breaks the index usage of an already problematic query. So should only be used with care

Reply:
@nickvergessen I'm confused, is the `affecteduser` the user for whom the activity is fired off or is the `affecteduser` the user the user who the activity is for?

If a user deleted an account and an activity for admins is fired, would the `affecteduser` row be the deleted username or the admin username? I want all the activities that are fired  for admin users to not expire, to maintain a history of important activities..

Reply:
User is the actor of the action,
Affected user is the one viewing the activity stream

But then it shows a bit that the config option needs better explanation.
But I still think that only 1 of the checks should stay

Reply:
> I want all the activities that are fired for admin users to not expire, to maintain a history of important activities..

Btw there is also the audit log feature in Nextcloud which is specifically designed for things like this:
https://docs.nextcloud.com/server/latest/developer_manual/basics/logging.html#admin-audit-logging

Reply:
@nickvergessen But then I need to add some new code or patch the files on server as admin audit logs these user deletion and other important actions at "INFO" level.. It uses the `loglevel` variable and if I allow "INFO" level logs, I'm afraid performance takes a hit for my server :(

Reply:
> should be only the affecteduser below.

@nickvergessen Done

Reply:
https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/logging_configuration.html#log-level-interaction